### PR TITLE
Fixed issue #17223: With mandatory attribute : lost all current value when create a token

### PR DESCRIPTION
**Pull request by gabrieljenik**
Adding JS validation for required token attributes

### CODE REVIEW

File: assets/scripts/admin/tokens.js [hunk 1]
```diff
@@ -174,6 +174,50 @@ function submitEditToken(){
...
 
+/**
+ * Validates that mandatory additional attributes are filled
+ */
+function validateAdditionalAttributes() {
+    const reportFieldValidity = function (field) {
+        // Pick the message from the field attributes, or use a default one if not set
+        var msg = field.attr('data-empty-msg');
+        if (!msg) {
+            msg = "Please fill out this field";
```
Comment (Shnoulle) [inline assets/scripts/admin/tokens.js:185]: ? Why ? Browser have own language string here.

**Comment by Shnoulle** [comment]
My opinion : better to use required attribute and when try to submit : go the 1st tab with error.

File: assets/scripts/admin/tokens.js (hunk 1 above)
Comment (gabrieljenik) [reply assets/scripts/admin/tokens.js]: Updated

**Comment by gabrieljenik** [comment]
Now validation and the way of showing the validation mssage has been simplified using a toast message. 

HTML5 standard validation message  or field highlighting is not used anymore as to simplify the code (could be error prone and complex to do that on hidden tabs)

This is a starter. If we want to make it more complex or enhance it, it can be done from here.

The validation has also been added for the edit process, which didn't perform any validation at all

**Comment by olleharstedt** [comment]
Please pull latest upstream to fix Github Actions.

**Comment by gabrieljenik** [comment]
Ready!

File: assets/scripts/admin/tokens.js [hunk 2]
```diff
@@ -174,6 +177,33 @@ function submitEditToken(){
...
+    $('#edittoken-error-container').show();
+}
+
+/**
+ * Validates that mandatory additional attributes are filled
+ */
+function validateAdditionalAttributes() {
+    const validationErrorMsg = $('#edittoken').attr('data-validation-error');
+
+    $valid = true;
```
Comment (olleharstedt) [inline assets/scripts/admin/tokens.js:191]: Hm, why is `$valid` with a dollar sign here? It's JS, not PHP. :)

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: assets/scripts/admin/tokens.js (hunk 2 above)
Comment (olleharstedt) [reply assets/scripts/admin/tokens.js:191]: Guess you want to use "let" or "var" as well.

**Pull request merged**